moped-editor/src/views/projects/projectsListView/ProjectsListViewTable.js

```diff
@@ -135,13 +144,21 @@ const ProjectsListViewTable = () => {
 
   const data = useCurrentData(projectListViewData);
 
-  const { handleExportButtonClick, dialogOpen } = useCsvExport({
+  const {
+    handleExportButtonClick,
+    handleRadioSelect,
+    handleContinueButtonClick,
+    handleOptionsDialogClose,
+  } = useCsvExport({
```

Comment:
i found this pattern of returning the functions from `useCsvExport` here and then using them later in the props of the dialog components in this file to be kinda weird but i just followed that pattern from what was here already ?

Reply:
I'm far from 100% on this, but FWIW, maybe because the arguments that need to go into that custom hook are only available in this component, and it's easier to drill down these handler callbacks than make so much of the state of `ProjectListViewTable` a prop to `CsvDownloadOptionsDialog. I'm curious too.

Reply:
Yes, this was 100% an attempt on my part to organize chunks of code that were being split out from a large component, and I agree that it feels weird and smells when looking at it now too. 👃

So, in `useCsvExport.js`, I see a component with no state, and a custom hook that contains all of the state and handlers needed for the UI in that component. I think that moving the contents of `useCsvExport` into `CsvDownloadOptionsDialog` would really cut down on the props passed around. I can expand if needed, and thanks for picking up on this!

Reply:
i wanted to update that Rose helped me see that what I outlined in my last comment wasn't as simple as I was thinking. As this code is growing, I think it could be good to pull the lazy query out of this hook along with the handlers and try to organize the state/components/handlers differently - but, I don't see that as a blocker to shipping this feature at all since this was an existing issue before this PR. 🚀